docs/Quiz Feature Planning.md

```diff
@@ -0,0 +1,311 @@
+
+    - **Request Payload:**
+        
+        ```
+        {
+          "question_id": "<question_id>",
+          "quiz_context": {
+                "question_text": "What is the purpose of the internet?",
+            "options": [
+                "Communication",
+                "Gaming",
+                "Education",
+                "All of the above"
+            ],
+            "correct_index": 3,
+            }
+        }
+        ```
+
+        - **Response Payload:**
+        ```
+        {
+            "explanation1": <exp1>,
+            "explanation2": <exp2>,
+            "explanation3": <exp3>,
+            "explanation4": <exp4>, 
+        }
+        ```
+
+
+
+3. **AI Service-check answer**
+
+ - **Endpoint:** `POST /check-answer`
+        
+    - **Description:** Validates the user's answer, determines correctness, and provides an explanation.
+
+### **Data Flow**
+
+1. User request a quiz or a question to be generated.
+
+2. Main Backend forwards the request to the quiz service.
+
+3. Quiz service generates the quiz/question along with explanation and stores it in database, returing it to the main backend. Or, quiz/question is retrieved from the database and returned to the main backend.
+
+4. Main Backend delivers the response to the extention to render.
+
+5. After user made the choice, extention show the result, and explanation for each option.
+
+## **Error Handling**
+
+1. **Failed to get Transcripts:**
+
+    -Notify the user about the transcript issue.
+
+
+## **Scalability and Performance Considerations**
+
+1. Use Redis for caching quizzes, transcipt for specific videos, so that when user comes back to the video or other users watch the same video can use the generated quizzes and questions.
+
+## **Display Plan**
+
+- Integrate the summary feature into the existing interface:
+
+    - **Extension:** Add a "Generate quiz/question" button in the video player controls.
+        
+    - **Web Interface:** Display summaries in a dedicated section below the video.
+
+    ---
+
+## **Architectural Decision Record (ADR)**
+
+### **Title:** Quiz Feature
+
+### **Status:**
+
+Proposed
+
+### **Context:**
+
+The Quiz feature aims to enhance user engagement by generating real-time quizzes that are relevant to the video.
+
+### **Decision:**
+
+Implement the whole quizzes generation. Quiz Service, retrieves the quiz if there's quiz existing for the associated video,or fetches the transcript, metadata, length of video from Info Service. Devide into windows if longer than 1 hour, and use AI Service to generate quiz and explanation for each opton for each window, combine if necessary. Finally, storing the quiz in redis and return to main backend and then the extention.
+
+![Quiz ADR](QuizDesign.png "Quiz ADR")
+
+### **Rationale:**
+
+- Easier to implement.
+
+- Easier to store and retrieve since timestamp doesnt matter.
+
+### **Trade-offs:**
+
+- If a user is watching a long video, the questions might not be instantly relevant.
```

Comment:
Maybe mention what we are doing for this, Since we are planning on chunking longer videos